application-task-default/src/main/java/com/xwiki/task/internal/TaskPageCopyEventListener.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.task.internal;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.xwiki.bridge.event.DocumentCreatedEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.job.event.JobStartedEvent;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.ObservationContext;
+import org.xwiki.observation.event.Event;
+import org.xwiki.refactoring.event.DocumentCopyingEvent;
+import org.xwiki.refactoring.event.DocumentRenamingEvent;
+import org.xwiki.refactoring.job.AbstractCopyOrMoveRequest;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.task.TaskCounter;
+import com.xwiki.task.TaskException;
+import com.xwiki.task.model.Task;
+
+import static com.xwiki.task.internal.AbstractTaskEventListener.TASK_CLASS_REFERENCE;
+
+/**
+ * This listener handles the case when a page containing a task object is copied. The id of the task object should be
+ * updated to a new one. No two tasks should have the same id.
+ *
+ * @version $Id$
+ * @since 3.5.2
+ */
+@Component
+@Named("TaskPageCopyEventListener")
+@Singleton
+public class TaskPageCopyEventListener extends AbstractEventListener
```

Comment:
I will rename it to `TaskPageMovingOrCopyingEventListener` since it reacts to both actions. (and it also matches `org.xwiki.refactoring.internal.job.AbstractCopyOrMoveJob`.